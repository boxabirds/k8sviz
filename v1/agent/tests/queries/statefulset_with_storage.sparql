# Query: Create a StatefulSet for a database with persistent storage
# This query retrieves information for generating a StatefulSet with PVC

PREFIX k8s: <http://kubernetes.io/ontology#>
PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>

SELECT ?resource ?property ?value ?comment ?constraint
WHERE {
  # Find StatefulSet resource and properties
  {
    k8s:StatefulSet a owl:Class ;
                    rdfs:comment ?statefulSetComment .
    
    ?property rdfs:domain k8s:StatefulSet ;
              rdfs:label ?propertyLabel ;
              rdfs:comment ?propertyComment .
  }
  
  # Get PersistentVolumeClaim requirements
  UNION
  {
    k8s:PersistentVolumeClaim a owl:Class ;
                              rdfs:comment ?pvcComment .
    
    ?pvcProperty rdfs:domain k8s:PersistentVolumeClaim ;
                 rdfs:label ?pvcPropertyLabel .
  }
  
  # Get storage access modes
  UNION
  {
    ?accessMode a k8s:AccessMode ;
                rdfs:label ?accessModeLabel ;
                rdfs:comment ?accessModeComment .
  }
  
  # Get storage constraints
  UNION
  {
    ?constraint a owl:Class ;
                rdfs:label ?constraintLabel ;
                rdfs:comment ?constraintComment .
    FILTER(CONTAINS(LCASE(STR(?constraintComment)), "storage") || 
           CONTAINS(LCASE(STR(?constraintComment)), "volume"))
  }
  
  # Get StatefulSet specific features
  UNION
  {
    SELECT ?feature ?featureValue
    WHERE {
      ?feature rdfs:domain k8s:StatefulSet ;
               owl:hasValue ?featureValue .
    }
  }
}