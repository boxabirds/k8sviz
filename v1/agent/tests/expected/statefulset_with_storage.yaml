apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: postgres-db
spec:
  serviceName: postgres-service
  replicas: 3
  selector:
    matchLabels:
      app: postgres
  template:
    metadata:
      labels:
        app: postgres
    spec:
      containers:
      - name: postgres
        image: postgres:14
        ports:
        - containerPort: 5432
          name: postgres
        env:
        - name: POSTGRES_PASSWORD
          valueFrom:
            secretKeyRef:
              name: postgres-secret
              key: password
        - name: PGDATA
          value: /var/lib/postgresql/data/pgdata
        volumeMounts:
        - name: postgres-storage
          mountPath: /var/lib/postgresql/data
        resources:
          requests:
            cpu: "500m"
            memory: "1Gi"
          limits:
            cpu: "1000m"
            memory: "2Gi"
        livenessProbe:
          exec:
            command:
            - /bin/sh
            - -c
            - pg_isready -U postgres
          initialDelaySeconds: 30
          periodSeconds: 10
        readinessProbe:
          exec:
            command:
            - /bin/sh
            - -c
            - pg_isready -U postgres
          initialDelaySeconds: 5
          periodSeconds: 5
  volumeClaimTemplates:
  - metadata:
      name: postgres-storage
    spec:
      accessModes: [ "ReadWriteOnce" ]
      storageClassName: "standard"
      resources:
        requests:
          storage: 10Gi